ayekan/exporters/snmp-generator/Makefile

```diff
@@ -0,0 +1,356 @@
+
+$(MIBDIR)/servertech-sentry4-mib:
+	@echo ">> Downloading servertech-sentry4-mib"
+	@curl $(CURL_OPTS) -o $(MIBDIR)/servertech-sentry4-mib $(SERVERTECH4_URL)
+
+$(MIBDIR)/.synology:
+	$(eval TMP := $(shell mktemp))
+	@echo ">> Downloading synology to $(TMP)"
+	@curl $(CURL_OPTS) -o $(TMP) $(SYNOLOGY_URL)
+	@unzip -j -d $(MIBDIR) $(TMP)
+	@rm -v $(TMP)
+	@touch $(MIBDIR)/.synology
+
+$(MIBDIR)/UBNT-UniFi-MIB:
+	@echo ">> Downloading UBNT-UniFi-MIB"
+	@curl $(CURL_OPTS) -o $(MIBDIR)/UBNT-UniFi-MIB "$(UBNT_DL_URL)/UBNT-UniFi-MIB"
+
+$(MIBDIR)/UBNT-AirFiber-MIB:
+	@echo ">> Downloading UBNT-AirFiber-MIB"
+	@curl $(CURL_OPTS) -o $(MIBDIR)/UBNT-AirFiber-MIB $(UBNT_AIRFIBER_URL)
+
+$(MIBDIR)/UBNT-AirMAX-MIB.txt:
+	$(eval TMP := $(shell mktemp))
+	@echo ">> Downloading ubnt-airos to $(TMP)"
+	@curl $(CURL_OPTS) -o $(TMP) $(UBNT_AIROS_URL)
+	@unzip -j -d $(MIBDIR) $(TMP) UBNT-AirMAX-MIB.txt
+	@rm -v $(TMP)
+
+$(MIBDIR)/PDU-MIB.txt:
+	@echo ">> Downloading Raritan PDU-MIB"
+	@curl $(CURL_OPTS) -o $(MIBDIR)/PDU-MIB.txt "$(RARITAN_URL)"
+
+$(MIBDIR)/PDU2-MIB.txt:
+	@echo ">> Downloading Raritan PDU2-MIB"
+	@curl $(CURL_OPTS) -o $(MIBDIR)/PDU2-MIB.txt "$(RARITAN2_URL)"
+
+$(MIBDIR)/Infrapower-MIB.mib:
+	$(eval TMP := $(shell mktemp))
+	@echo ">> Downloading Infrapower-MIB.mib to $(TMP)"
+	@rm -vf $(MIBDIR)/IPD-MIB_Q419V9.mib
+	@curl $(CURL_OPTS) -L -o $(TMP) $(INFRAPOWER_URL)
+	@unzip -j -d $(MIBDIR) $(TMP) IPD-03-S-MIB_Q320V1.mib
+	@mv -v $(MIBDIR)/IPD-03-S-MIB_Q320V1.mib $(MIBDIR)/Infrapower-MIB.mib
+	@rm -v $(TMP)
+
+$(MIBDIR)/LIEBERT_GP_PDU.MIB:
+	$(eval TMP := $(shell mktemp))
+	@echo ">> Downloading LIEBERT_GP_PDU.MIB to $(TMP)"
+	@curl $(CURL_OPTS) -o $(TMP) $(LIEBERT_URL)
+	@unzip -j -d $(MIBDIR) $(TMP) LIEBERT_GP_PDU.MIB LIEBERT_GP_REG.MIB
+	@rm -v $(TMP)
+
+$(MIBDIR)/CyberPower.MIB:
+	$(eval TMP := $(shell mktemp))
+	@echo ">> Downloading CyberPower.MIB to $(TMP)"
+	@curl $(CURL_OPTS) -o $(TMP) $(CYBERPOWER_URL)
+	@unzip -j -d $(MIBDIR) $(TMP) CyberPower_MIB_v$(CYBERPOWER_VERSION).MIB
+	@mv -v $(MIBDIR)/CyberPower_MIB_v$(CYBERPOWER_VERSION).MIB $@
+	# Workaround to make DisplayString available (#867)
+	@sed -i.bak -E 's/(DisplayString[[:space:]]*FROM )RFC1213-MIB/\1SNMPv2-TC/' $@
+	@rm $@.bak
+	@rm -v $(TMP)
+
+$(MIBDIR)/EAP.MIB $(MIBDIR)/EAP-Client.MIB:
+	$(eval TMP := $(shell mktemp))
+	@echo ">> Downloading EAP.MIB to $(TMP)"
+	@curl $(CURL_OPTS) -o $(TMP) $(EAP_URL)
+	@unzip -j -d $(MIBDIR) $(TMP) EAP_Private-Mibs_$(EAP_VERSION)/TPLINK.mib EAP_Private-Mibs_$(EAP_VERSION)/EAP/clientMib/client.mib
+	@mv -v $(MIBDIR)/TPLINK.mib $(MIBDIR)/EAP.MIB
+	@mv -v $(MIBDIR)/client.mib $(MIBDIR)/EAP-Client.MIB
+	@rm -v $(TMP)
+
+$(MIBDIR)/readynas:
+	@echo ">> Downloading readynas"
+	@curl $(CURL_OPTS) -o $(MIBDIR)/readynas "$(READYNAS_URL)"
+
+$(MIBDIR)/readydataos:
+	@echo ">> Downloading readydataos"
+	@curl $(CURL_OPTS) -o $(MIBDIR)/readydataos "$(READYDATAOS_URL)"
+
+$(MIBDIR)/XUPS-MIB:
```

Comment:
The target should be `$(MIBDIR)/XUPS-MIB.mib` to prevent the mib from being re-downloaded on every build.